### ISSUE TITLE
Weird behaviour while linting dbt module

### ISSUE THREAD

**Issue by kostek-pl** [sqlfluff/sqlfluff#1701]
## Expected Behaviour
L003 | Indentation not consistent

## Observed Behaviour
No errors

#### Step 1 - running dbt model(we should see "Indentation not consistent")
```sql
{{
    config(
        materialized = "incremental"
    )
}}

with test as (

    select
        aaa,
         bbb,
          ccc
    from
        {{ source('src', 'table1') }}
    where
        ccc is not null
)

select
    aaa,
    bbb,
    ccc
from
    test
```
```shell
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
All Finished 📜 🎉!
```

#### Step 2 - changing  {{ source('src', 'table1') }} ---> src.table1
```sql
{{
    config(
        materialized = "incremental"
    )
}}

with test as (

    select
        aaa,
         bbb,
          ccc
    from
        src.table1
    where
        ccc is not null
)

select
    aaa,
    bbb,
    ccc
from
    test
```

```shell
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [sqlfluff_test.sql] FAIL
L:  10 | P:  14 | L003 | Indentation not consistent with line #6
L:  11 | P:  15 | L003 | Indentation not consistent with line #6
```
#### Step 3 - removing configuration part
```sql
with test as (

    select
        aaa,
         bbb,
          ccc
    from
        {{ source('src', 'table1') }}
    where
        ccc is not null
)

select
    aaa,
    bbb,
    ccc
from
    test
```
```shell
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [sqlfluff_test.sql] FAIL
L:   5 | P:   6 | L003 | Indentation not consistent with line #4
L:   6 | P:   6 | L003 | Indentation not consistent with line #4
All Finished 📜 🎉!
```

## Dialect
redshift/postgres

## Version
0.7.0  
3.8.7

## Configuration
```
[sqlfluff]
templater = dbt
dialect = postgres
nocolor = True

[sqlfluff:templater:dbt]
project_dir = .\test
```

**Comment by barrywhart** [comment]
This is working fine for me. Lint output:
```
(sqlfluff-3.7.9) ➜  my-dbt sqlfluff lint models/issue_1701.sql
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/issue_1701.sql] FAIL
L:  11 | P:  10 | L003 | Indentation not consistent with line #6
L:  12 | P:  11 | L003 | Indentation not consistent with line #6
```

My dbt setup is slightly different. I'm using a BigQuery profile. This is my `.sqlfluff` file. I tested this using the 0.7.0 tag and the latest code in `main`.

```
[sqlfluff]
dialect = bigquery
templater = dbt
```

In case it matters, these are the versions dbt-related packages I'm using:
```
dbt==0.19.1
dbt-bigquery==0.19.1
dbt-core==0.19.1
dbt-postgres==0.19.1
dbt-redshift==0.19.1
dbt-snowflake==0.19.1
```

@kostek-pl: Can you test this again and/or provide more detail to help us reproduce the behavior you're seeing? Perhaps providing a full zipped-up dbt project would be useful here?

**Comment by barrywhart** [comment]
No response received. Cannot reproduce. Closing.
